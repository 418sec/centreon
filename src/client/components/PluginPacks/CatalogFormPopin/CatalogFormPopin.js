import React, { Component } from 'react'
import { withStyles } from 'material-ui/styles'
import Dialog, { DialogActions, DialogContent, DialogTitle } from 'material-ui/Dialog';
import Radio, { RadioGroup } from 'material-ui/Radio';
import { FormControlLabel } from 'material-ui/Form';
import Button from 'material-ui/Button';

const styles = {

}

class CatalogFormPopin extends Component {
  render = () => {
    const { level, handleChange, handleCancel, handleConfirm, ...other } = this.props

    return (
      <Dialog
        disableBackdropClick
        disableEscapeKeyDown
        maxWidth="xs"
        {...other}
      >
        <DialogTitle>Choose catalog access</DialogTitle>
        <DialogContent>
          <RadioGroup
            aria-label="level"
            name="level"
            value={level}
            onChange={handleChange}
          >
            <FormControlLabel value={'1'} key={1} control={<Radio />} label={'1 : Free'} />
            <FormControlLabel value={'2'} key={2} control={<Radio />} label={'2 : Registered'} />
            <FormControlLabel value={'3'} key={3} control={<Radio />} label={'3 : Subscribed'} />
          </RadioGroup>
        </DialogContent>
        <DialogActions>
          <Button id="cancelCatalogFormPopin" onClick={handleCancel} color="primary">
            Cancel
          </Button>
          <Button id="confirmCatalogFormPopin" onClick={handleConfirm} color="primary">
            Confirm
          </Button>
        </DialogActions>
      </Dialog>
    )
  }
}

export default withStyles(styles)(CatalogFormPopin)