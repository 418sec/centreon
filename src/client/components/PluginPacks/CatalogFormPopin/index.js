export { default as CatalogFormPopin } from './CatalogFormPopinContainer';