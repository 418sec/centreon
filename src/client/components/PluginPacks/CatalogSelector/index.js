export { default as CatalogSelector } from './CatalogSelectorContainer';