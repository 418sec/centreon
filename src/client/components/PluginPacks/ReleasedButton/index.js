export { default as ReleasedButton } from './ReleasedButtonContainer';