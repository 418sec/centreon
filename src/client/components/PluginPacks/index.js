export { default as PluginPacksList } from './PluginPacksListContainer';