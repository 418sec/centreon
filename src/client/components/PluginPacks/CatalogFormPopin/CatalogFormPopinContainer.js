import React, { Component } from 'react'
import { connect } from 'react-redux'
import CatalogFormPopin from './CatalogFormPopin'
import { patchPluginPack } from 'Services/hubApi'
import { closeCatalogFormPopin } from 'Redux/pluginPacks/pluginPacksActions'

class CatalogFormPopinContainer extends Component {
  constructor (props) {
    super(props)

    this.state = {
      open: props.open,
      level: props.level
    }
  }

  componentWillReceiveProps = (nextProps) => {
    this.setState({
      open: nextProps.open,
      level: nextProps.level
    })
  }

  handleChange = (event, level) => {
    this.setState({ level: level })
  }

  handleCancel = () => {
    this.props.closeCatalogFormPopin()
  }

  handleConfirm = () => {
    const {token, id} = this.props
    const { level } = this.state

    this.props.closeCatalogFormPopin()
    this.props.patchPluginPack(token, id, level)
  }

  render = () => {
    const { open, level } = this.state

    return (
      <CatalogFormPopin
        handleChange={this.handleChange}
        handleCancel={this.handleCancel}
        handleConfirm={this.handleConfirm}
        open={open}
        level={level}
      />
    )
  }

}

const mapStateToProps = (store) => {
  return {
    token: store.authReducer.token,
    open: store.pluginPacksReducer.catalogFormPopin.open,
    id: store.pluginPacksReducer.catalogFormPopin.id,
    level: store.pluginPacksReducer.catalogFormPopin.level,
  }
}


const mapDispatchToProps = (dispatch) => {
  return {
    patchPluginPack: (token, pluginPackId, attributes) => {
      return dispatch(patchPluginPack(token, pluginPackId, attributes))
    },
    closeCatalogFormPopin: () => {
      return dispatch(closeCatalogFormPopin())
    }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(CatalogFormPopinContainer)