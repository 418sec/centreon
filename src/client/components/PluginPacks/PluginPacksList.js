import React, { Component } from 'react'
import {
  PagingState,
  FilteringState,
  CustomPaging
} from '@devexpress/dx-react-grid'
import {
  Grid,
  Table,
  TableHeaderRow,
  TableFilterRow,
  PagingPanel
} from '@devexpress/dx-react-grid-material-ui'
import PropTypes from 'prop-types'
import { withStyles } from 'material-ui/styles'
import { TableRow, TableCell } from 'material-ui'
import { HeaderBar } from 'Components/HeaderBar'
import { CatalogFormPopin } from 'Components/PluginPacks/CatalogFormPopin'
import { CatalogSelector } from 'Components/PluginPacks/CatalogSelector'
import { ReleasedButton } from 'Components/PluginPacks/ReleasedButton'
import { FilterCell } from 'Components/PluginPacks/FilterCell'

const styles = theme => ({
  root: {
    padding: '0 15px',
    overflow: 'auto'
  },
  TableCell: {
    wordWrap: 'break-word',
    padding: '4px 14px',
  },
  check: {
    fill: '#88B917'
  },
  checkInactive: {
    fill: '#d1d2d4'
  },
  textField: {
    '&:hover': {
      background:'transparent'
    },
  }
})

class PluginPacksList extends Component {

  tableRowTemplate = ({children, row}) => {
    return (
      <TableRow id={row.slug}>
        {children}
      </TableRow>
    )
  }

  render = () => {
    const { columns, rows, itemsPerPage, totalItems, currentPage, classes } = this.props

    return (
      <section>
        <div id="pluginPackList" className={classes.root}>
          <Grid
            columns={columns}
            rows={rows}
          >
            <FilteringState
              onFiltersChange={this.props.changeFilters}
            />
            <PagingState
              currentPage={currentPage}
              onCurrentPageChange={this.props.changeCurrentPage}
              pageSize={itemsPerPage}
            />
            <CustomPaging
              totalCount={totalItems}
            />
            <Table
              rowComponent={this.tableRowTemplate}
              cellComponent={({ column, row, value }) => {
                if (column.name === 'released') {
                  return (
                    <TableCell>
                      <ReleasedButton
                        pluginPackId={row.id}
                      />
                    </TableCell>
                  )
                } else if (column.name === 'catalog_level') {
                  return (
                    <TableCell>
                      <CatalogSelector
                        id={row.id}
                        level={value}
                      />
                    </TableCell>
                  )
                }

                return (
                  <TableCell className={classes.TableCell}>{value}</TableCell>
                )
              }}
            />
            <TableHeaderRow />
            <PagingPanel id={"pagingPanel"}/>
            <TableFilterRow
              messages={{ filterPlaceholder: 'Filter' }}
              cellComponent={FilterCell}
            />
          </Grid>
        </div>
        <CatalogFormPopin />
      </section>
    )
  }
}

/**
 * Instance list Prototypes
 */
PluginPacksList.propTypes = {
  /**
   * Instance list css classes
   * @param {Object} classes
   */
  classes: PropTypes.object.isRequired,
}

PluginPacksList.defaultProps = {
  currentPage: 1
}

export default withStyles(styles)(PluginPacksList)
