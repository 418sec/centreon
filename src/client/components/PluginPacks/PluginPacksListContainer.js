import React, { Component } from 'react'
import { withRouter } from 'react-router-dom'
import { connect } from 'react-redux'
import PluginPacksList from './PluginPacksList'
import { getPluginPacks } from 'Services/hubApi'
import { cleanHeader, replaceTitle} from 'Redux/header/headerActions'

class PluginPacksListContainer extends Component {

  constructor (props) {
    super(props)

    this.state = {
      columns: [
        {name: 'slug', title: 'Slug'},
        {name: 'name', title: 'Name'},
        {name: 'version', title: 'Version'},
        {name: 'catalog_level', title: 'Catalog'},
        {name: 'released', title: 'Released'},
      ],
      currentPage: 0,
      itemsPerPage: 10,
      filters: []
    }
  }

  componentDidMount = () => {
    const { cleanHeader, replaceHeaderTitle } = this.props
    cleanHeader()
    replaceHeaderTitle('Plugin Packs')

    this.getData()
  }

  changeFilters = (filters) => {
    this.setState(
      {
        filters: filters,
        currentPage: 0
      },
      this.getData
    )
  }

  changeCurrentPage = (currentPage) => {
    this.setState(
      {currentPage: currentPage},
      this.getData
    )
  }

  /**
   * example : page[number]=1&page[size]=30
   * @returns {string|*}
   */
  getPagination = () => {
    const { currentPage, itemsPerPage } = this.state
    return `?page[number]=${currentPage + 1}&page[size]=${itemsPerPage}`
  }

  /**
   * example : filter[instanceName]=centreon&filter[fingerprint]=abc
   * @returns {string|*}
   */
  getFilters = () => {
    const { filters } = this.state;

    return filters.reduce((acc, { columnName, value }) => {
      if (columnName == 'catalog_level' && value == '0') {
        acc.push(`filter[with_catalog]=false`)
      } else {
        acc.push(`filter[${columnName}]=${value}`)
      }
      return acc
    }, []).join('&')
  }

  getData = () => {
    const { token, displayPluginPacks } = this.props
    displayPluginPacks(this.getPagination(), this.getFilters(), token)
  }

  render = () => {
    const { columns, currentPage } = this.state
    const { items, itemsPerPage, totalPages, totalItems, ...props } = this.props

    const rows = items ?
      Object.keys(items).reduce((acc, key) => {
        acc[items[key].order] = items[key]
        return acc
      }, [])
      : []

    return (
      <PluginPacksList
        columns={columns}
        rows={rows}
        itemsPerPage={itemsPerPage}
        totalPages={totalPages}
        totalItems={totalItems}
        currentPage={currentPage}
        changeFilters={this.changeFilters}
        changeCurrentPage={this.changeCurrentPage}
        {...props}
      />
    )
  }
}

const mapStateToProps = (store) => {
  return {
    token: store.authReducer.token,
    items: store.pluginPacksReducer.items,
    totalPages: store.pluginPacksReducer.totalPages,
    totalItems: store.pluginPacksReducer.totalItems,
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    cleanHeader: () => {
      return dispatch(cleanHeader())
    },
    replaceHeaderTitle: (title) => {
      return dispatch(replaceTitle(title))
    },
    displayPluginPacks: (pagination, filters, token) => {
      return dispatch(getPluginPacks(pagination, filters, token))
    },
  }
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(PluginPacksListContainer))