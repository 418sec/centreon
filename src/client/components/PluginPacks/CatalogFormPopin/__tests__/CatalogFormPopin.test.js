/* eslint-env jest */
import React from 'react'
import Enzyme from 'enzyme'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
import configureStore from 'redux-mock-store'

Enzyme.configure({ adapter: new Adapter() })

// add mock for hub api
jest.mock('Services/hubApi', () => ({
  patchPluginPack: () => ({
    type: 'PATCH_PLUGIN_PACK_SUCCESS',
    catalogFormPopin: {
      open: false
    },
    item: {}
  })
}))
import { CatalogFormPopin } from 'Components/PluginPacks/CatalogFormPopin'

describe('CatalogFormPopin', () => {
  let mockStore = configureStore()
  let store
  let component

  it('renders correctly', () => {
    store = mockStore({
      authReducer: {
        token: ''
      },
      pluginPacksReducer: {
        catalogFormPopin: {
          open: false,
          id: null,
          level: null
        }
      }
    })
    component = shallow(
      <CatalogFormPopin
        store={store}
      />
    ).dive().dive().dive()
    expect(toJson(component)).toMatchSnapshot()
  })

  it('closes correctly', () => {
    let actions

    store = mockStore({
      authReducer: {
        token: ''
      },
      pluginPacksReducer: {
        catalogFormPopin: {
          open: true,
          id: 1,
          level: '2'
        }
      }
    })
    component = mount(
      <CatalogFormPopin
        store={store}
      />
    )
    component.find('button#confirmCatalogFormPopin').simulate('click')
    actions = store.getActions()
    expect(actions[0].type).toBe('CLOSE_CATALOG_FORM_POPIN')
  })
})