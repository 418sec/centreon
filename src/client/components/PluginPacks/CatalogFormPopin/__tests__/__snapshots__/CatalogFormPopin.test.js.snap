// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatalogFormPopin renders correctly 1`] = `
<WithStyles(Dialog)
  classes={Object {}}
  disableBackdropClick={true}
  disableEscapeKeyDown={true}
  maxWidth="xs"
  open={false}
>
  <WithStyles(DialogTitle)>
    Choose catalog access
  </WithStyles(DialogTitle)>
  <WithStyles(DialogContent)>
    <RadioGroup
      aria-label="level"
      name="level"
      onChange={[Function]}
      value={null}
    >
      <WithStyles(FormControlLabel)
        control={<WithStyles(Radio) />}
        key="1"
        label="1 : Free"
        value="1"
      />
      <WithStyles(FormControlLabel)
        control={<WithStyles(Radio) />}
        key="2"
        label="2 : Registered"
        value="2"
      />
      <WithStyles(FormControlLabel)
        control={<WithStyles(Radio) />}
        key="3"
        label="3 : Subscribed"
        value="3"
      />
    </RadioGroup>
  </WithStyles(DialogContent)>
  <WithStyles(DialogActions)>
    <WithStyles(Button)
      color="primary"
      id="cancelCatalogFormPopin"
      onClick={[Function]}
    >
      Cancel
    </WithStyles(Button)>
    <WithStyles(Button)
      color="primary"
      id="confirmCatalogFormPopin"
      onClick={[Function]}
    >
      Confirm
    </WithStyles(Button)>
  </WithStyles(DialogActions)>
</WithStyles(Dialog)>
`;
